### PR TITLE
Replacing sys.argv with argparse

### PR DESCRIPTION
**Pull request by aewens**
This is in regards to issue #2 and will provide automated help dialog along with a better framework to add new arguments in the future.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -25,16 +26,19 @@
...
-        usr = argv[1]
+    parser = ArgumentParser(prog="gps.py")
+    parser.add_argument("profile_name", dest="user", nargs="?", type=str,
+        help="Your GitHub profile name")
+    args = parser.parse_args()
+    user = args.user
+    if user is None:
+        user = input("Profile's name: ")
+        
+    if len(user) == 0:
```
Comment (pBouillon) [inline src/main.py:37]: wouldn't 
```python
if not user:
    # ...
```
do the same ?

**Review by pBouillon — COMMENTED** [review]
Thank you it's indeed better ! May I just suggest this readability improvement ?

File: src/main.py (hunk 1 above)
Comment (aewens) [reply src/main.py:37]: As opposed to `if user is None:`? Sure, that would technically do the same thing. I'll have that changed in just a moment.

Comment (aewens) [reply src/main.py:37]: This has been changed in commit [56d62b0](https://github.com/aewens/gps/commit/56d62b0d1af1f766416c8b3ac2e44c9f1512cce6#diff-0c81b9c94c68ea34c497c1fa52daee56)

**Comment by pBouillon** [comment]
All good, thank you ! 😄

**Pull request merged**